src/components/ZetkinCalendar/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { Box, Button, MenuItem, TextField, Typography } from '@material-ui/core';
+import { FormattedDate, FormattedMessage as Msg } from 'react-intl';
+
+import MonthCalendar from './MonthCalendar';
+import { useFocusDate } from '../../hooks';
+import { useIntl } from 'react-intl';
+import WeekCalendar from './WeekCalendar';
+import { ZetkinCampaign, ZetkinEvent, ZetkinTask } from '../../types/zetkin';
+
+enum CALENDAR_RANGES {
+    MONTH = 'month',
+    WEEK = 'week',
+}
+
+interface ZetkinCalendarProps {
+    baseHref: string;
+    events: ZetkinEvent[];
+    campaigns: ZetkinCampaign[];
+    tasks: ZetkinTask[];
+}
+
+const ZetkinCalendar = ({ baseHref, events, campaigns , tasks }: ZetkinCalendarProps) : JSX.Element => {
+    const { orgId } = useRouter().query;
+    const intl = useIntl();
+    const { focusDate, setFocusDate } = useFocusDate();
+    const [range, setRange] = useState(CALENDAR_RANGES.MONTH);
+
+    const sortedEvents = [...events].sort((a, b) => {
+        return new Date(a.start_time).getTime() - new Date(b.start_time).getTime();
+    });
+
+    const filteredTasks = tasks.filter(task => task.deadline);
```

Comment:
Small thing but i think it's good to name these like `tasksWithDeadlines` so it's clearer. Just a suggestion!